File: src/main/java/liqp/nodes/AttributeNode.java
package liqp.nodes;

import com.google.common.collect.ImmutableList;
import java.util.List;
import lombok.Getter;

@Getter
public class AttributeNode implements LNode {

    private final LNode key;
    private final LNode value;

    public AttributeNode(LNode key, LNode value) {
        this.key = key;
        this.value = value;
    }

    @Override
    public Object render(RenderContext context) {
        return new Object[]{
                key.render(context),
                value.render(context)
        };
    }

  @Override
  public List<LNode> children() {
    return ImmutableList.of(key, value);
  }
}

File: liqp-core/src/main/kotlin/liqp/nodes/AttributeNode.kt
package liqp.nodes

import liqp.context.LContext
import liqp.node.LNode
import lombok.Getter

class AttributeNode(val key: LNode, val value: LNode) : LNode() {

  override val children: List<LNode> get() = listOf(key, value)

  override fun render(context: LContext): Any? {
    return arrayOf(key.render(context), value.render(context))
  }
}
